roles/grafana_agent/tasks/main.yml

```
@@ -0,0 +1,53 @@
+---
+- name: Include secrets
+  include_vars: "{{ secrets_path | mandatory }}/mimir_password.yml"
+  no_log: true
+  tags:
+    - always
+
+- name: "Import Grafana GPG key"
+  become: true
+  ansible.builtin.apt_key:
+    url: "https://apt.grafana.com/gpg.key"
+    state: "present"
+  when: ansible_pkg_mgr == "apt"
+
+- name: "Add Grafana's repository to APT sources list"
+  become: true
+  ansible.builtin.apt_repository:
+    repo: "deb https://apt.grafana.com stable main"
+    state: present
+  when: ansible_pkg_mgr == "apt"
+
+- name: "Add Grafana's repository to yum/dnf systems"
+  become: true
+  ansible.builtin.yum_repository:
+    baseurl: "https://rpm.grafana.com"
+    name: "grafana"
+    description: "Grafana's repository"
```

Please enable GPG checking:
```suggestion
    description: "grafana"
    gpgcheck: true
    gpgkey: https://rpm.grafana.com/gpg.key
```